### DISCUSSION TITLE
Xdvipdfmx doesn’t find .pro files

### DISCUSSION THREAD

**Discussion by simonp0420** [tectonic-typesetting/tectonic#1021] [Q&A]
This is a duplicate of [this](https://tectonic.newton.cx/t/xdvipdfmx-doesnt-find-pro-files/342) discourse post.  I didn't realize when I posted to `tectonic.newton.cx` that it was being sunsetted in favor of these discussion issues.  Anyway...

I’m trying to get tectonic to compile a complex document that uses a lot of diagrams generated with pstricks. The document can be successfully compiled with standard XeTeX.  To proceed, I’ve been adding .sty and .tex files from the pstricks package next to my main document incrementally, as tectonic complains that it doesn’t have them. It’s now to the point where it has completed running through the document with the XeTeX compiler, and is running xdvipdfmx, which complains that it can’t find, e.g. pstricks.pro (among others), even though I have also placed these .pro files in the same directory. Also, I have texlive installed, and kpsewhich finds these files in, e.g. /usr/share/texmf-dist/dvips/pstricks/pstricks.pro . I suppose that tectonic has its own custom version of xdvipdfmx which doesn’t use kpsewhich. Can anyone advise how to make tectonic aware of these files?

**Comment by vlasakm** [comment]
Can you please provide an ["MWE"](https://tex.meta.stackexchange.com/questions/228/ive-just-been-asked-to-write-a-minimal-working-example-mwe-what-is-that)? It would allow me to try the example and take a look / replicate the issue.

**Comment by simonp0420** [comment]
Thanks for your willingness to look at this.  I've placed the MWE in [this](https://github.com/simonp0420/tectonic_pstricks_mwe) github repo, because there are a lot of files in the src directory from the pstricks package.  

Having pared down the huge document to a MWE, I'm no longer seeing the same error.  Now, instead of complaining about not finding the .pro files, tectonic now complains that it doesn't understand the token "load" as shown below:

```
~/tectonic/tectonic_pstricks_mwe main > tectonic -X build

note: "version 2" Tectonic command-line interface activated
Running TeX ...
Rerunning TeX because "mwe.aux" changed ...
Running xdvipdfmx ...
warning: Unknown token "load"
warning: Interpreting PS code failed!!! Output might be broken!!!
warning: Interpreting special command ps: (ps:) failed.
warning: >> at page="1" position="(133.768, 667.198)" (in PDF)
warning: >> xxx "ps: /pssetRGBcolor /setrgbcolor load def /pssetCMYKcolor /se..."
warning: >> Reading special command stopped around >>def /pssetCMYKcolor /setcmykcolor load def /pssetGraycolor /...<<
warning: Unknown token "tx@Dict"
warning: Interpreting PS code failed!!! Output might be broken!!!
warning: Interpreting special command ps: (ps:) failed.
warning: >> at page="1" position="(133.768, 667.198)" (in PDF)
warning: >> xxx "ps:tx@Dict begin  gsave STV CP T /ps@rot 0 def grestore  end"
warning: >> Reading special command stopped around >>begin  gsave STV CP T /ps@rot 0 def grestore  end<<
warning: Unknown token "tx@Dict"
warning: Interpreting PS code failed!!! Output might be broken!!!
warning: Interpreting special command ps: (ps:) failed.
warning: >> at page="1" position="(133.768, 667.198)" (in PDF)
warning: >> xxx "ps:tx@Dict begin  gsave STV CP T /ps@refangle 0.  def gresto..."
warning: >> Reading special command stopped around >>begin  gsave STV CP T /ps@refangle 0.  def grestore  end<<
warning: Unknown token "tx@Dict"
...
```
